sqlmesh/core/state_sync/db/snapshot.py

```diff
@@ -166,111 +168,158 @@ def get_expired_snapshots(
         self,
         environments: t.Iterable[Environment],
         current_ts: int,
-        ignore_ttl: bool = False,
-    ) -> t.List[SnapshotTableCleanupTask]:
-        """Aggregates the id's of the expired snapshots and creates a list of table cleanup tasks.
-
-        Expired snapshots are snapshots that have exceeded their time-to-live
-        and are no longer in use within an environment.
-
-        Returns:
-            The set of expired snapshot ids.
-            The list of table cleanup tasks.
-        """
-        all_cleanup_targets = []
-        for _, cleanup_targets in self._get_expired_snapshots(
-            environments=environments,
-            current_ts=current_ts,
-            ignore_ttl=ignore_ttl,
-        ):
-            all_cleanup_targets.extend(cleanup_targets)
-        return all_cleanup_targets
-
-    def _get_expired_snapshots(
-        self,
-        environments: t.Iterable[Environment],
-        current_ts: int,
-        ignore_ttl: bool = False,
-    ) -> t.Iterator[t.Tuple[t.Set[SnapshotId], t.List[SnapshotTableCleanupTask]]]:
-        expired_query = exp.select("name", "identifier", "version").from_(self.snapshots_table)
+        ignore_ttl: bool,
+        batch_range: ExpiredBatchRange,
+    ) -> t.Optional[ExpiredSnapshotBatch]:
+        expired_query = exp.select("name", "identifier", "version", "updated_ts").from_(
+            self.snapshots_table
+        )
 
         if not ignore_ttl:
             expired_query = expired_query.where(
                 (exp.column("updated_ts") + exp.column("ttl_ms")) <= current_ts
             )
 
-        expired_candidates = {
-            SnapshotId(name=name, identifier=identifier): SnapshotNameVersion(
-                name=name, version=version
-            )
-            for name, identifier, version in fetchall(self.engine_adapter, expired_query)
-        }
-        if not expired_candidates:
-            return
+        # Use expanded tuple comparisons for cross-engine compatibility
+        # Native tuple comparisons like (a, b) > (x, y) don't work reliably across all SQL engines
+        columns = [
+            exp.column("updated_ts"),
+            exp.column("name"),
+            exp.column("identifier"),
+        ]
+        start_values = [
+            exp.Literal.number(batch_range.start.updated_ts),
+            exp.Literal.string(batch_range.start.name),
+            exp.Literal.string(batch_range.start.identifier),
+        ]
+
+        start_condition = _expanded_tuple_comparison(columns, start_values, exp.GT)
```

Comment:
This seems like something that could've been a part of `RowBoundary` itself. Or at the very least returning a list of literals

Reply:
Sure I felt that way too but thought it would get pushback. will update.